nav2_localization/src/nav2_localization.cpp

```diff
@@ -0,0 +1,242 @@
+
+#include "nav2_localization/nav2_localization.hpp"
+#include "nav2_util/string_utils.hpp"
+#include "tf2_ros/create_timer_ros.h"
+
+using namespace std::chrono_literals;
+
+namespace nav2_localization
+{
+
+LocalizationServer::LocalizationServer()
+: LifecycleNode("localization_server", "", true),
+  sample_motion_model_loader_("nav2_localization", "nav2_localization_base::SampleMotionModel"),
+  default_sample_motion_model_id_("DummyMotionSampler"),
+  matcher2d_loader_("nav2_localization", "nav2_localization_base::Matcher2D"),
+  default_matcher2d_id_("DummyMatcher2D"),
+  solver_loader_("nav2_localization", "nav2_localization_base::Solver"),
+  default_solver_id_("DummySolver")
+{
+    RCLCPP_INFO(get_logger(), "Creating localization server");
+
+    declare_parameter("sample_motion_model_id", default_sample_motion_model_id_);
+    declare_parameter("matcher2d_id", default_matcher2d_id_);
+    declare_parameter("solver_id", default_solver_id_);
+    declare_parameter("first_map_only", true);
+    declare_parameter("laser_scan_topic_", "scan");
+    declare_parameter("odom_frame_id", "odom");
+    declare_parameter("alpha1", 0.2);
+    declare_parameter("alpha2", 0.2);
+    declare_parameter("alpha3", 0.2);
+    declare_parameter("alpha4", 0.2);
+    declare_parameter("alpha5", 0.2);
+}
+
+LocalizationServer::~LocalizationServer()
+{
+    RCLCPP_INFO(get_logger(), "Destroying");
+}
+
+nav2_util::CallbackReturn
+LocalizationServer::on_configure(const rclcpp_lifecycle::State & state)
+{
+    RCLCPP_INFO(get_logger(), "Configuring localization interface");
+
+    initParameters();
+    initTransforms();
+    initMessageFilters();
+    initPubSub();
+    initPlugins();
+
+    return nav2_util::CallbackReturn::SUCCESS;
+}
+
+nav2_util::CallbackReturn
+LocalizationServer::on_cleanup(const rclcpp_lifecycle::State & state)
+{
+    initial_pose_sub_.reset();
+
+    // Laser Scan
+    laser_scan_connection_.disconnect();
+    laser_scan_filter_.reset();
+    laser_scan_sub_.reset();
+
+    // Transforms
+    tf_broadcaster_.reset();
+    tf_listener_.reset();
+    tf_buffer_.reset();
+
+    return nav2_util::CallbackReturn::SUCCESS;
+}
+
+nav2_util::CallbackReturn
+LocalizationServer::on_activate(const rclcpp_lifecycle::State & state)
+{
+    // Keep track of whether we're in the active state. We won't
+    // process incoming callbacks until we are
+    active_ = true;
+
+    return nav2_util::CallbackReturn::SUCCESS;
+}
+
+nav2_util::CallbackReturn
+LocalizationServer::on_error(const rclcpp_lifecycle::State &)
+{
+    RCLCPP_FATAL(get_logger(), "Lifecycle node entered error state");
+    return nav2_util::CallbackReturn::SUCCESS;
+}
+
+nav2_util::CallbackReturn
+LocalizationServer::on_shutdown(const rclcpp_lifecycle::State &)
+{
+    RCLCPP_INFO(get_logger(), "Shutting down");
+    return nav2_util::CallbackReturn::SUCCESS;
+}
+
+void
+LocalizationServer::initParameters()
+{
+    get_parameter("sample_motion_model_id", sample_motion_model_id_);
```

Comment:
Ideally the list of these should be short for this since we're using plugins so the application-level params can be found directly in `on_configure` since there's so many fewer of them now